include/vkwrap/vkwrap/pipeline_cfgs.h

```diff
@@ -0,0 +1,352 @@
+    BlendStateCfg() = default;
+    BlendStateCfg( BlendStateCfg&& ) = default;
+    BlendStateCfg( const BlendStateCfg& ) = default;
+    BlendStateCfg& operator=( const BlendStateCfg& ) = default;
+    BlendStateCfg& operator=( BlendStateCfg&& ) = default;
+    ~BlendStateCfg() = default;
+
+  private:
+    vk::PipelineColorBlendAttachmentState m_color_blend_attachment;
+    vk::PipelineColorBlendStateCreateInfo m_color_blending;
+};
+
+template <typename Derived> class RenderPassCfg
+{
+  public:
+    Derived& withColorAttachment( vk::Format swap_chain_format ) &
+    {
+        m_color_attachment.setFormat( swap_chain_format );
+        m_color_attachment.setSamples( vk::SampleCountFlagBits::e1 );
+        m_color_attachment.setLoadOp( vk::AttachmentLoadOp::eClear );
+        m_color_attachment.setStoreOp( vk::AttachmentStoreOp::eStore );
+        m_color_attachment.setStencilLoadOp( vk::AttachmentLoadOp::eDontCare );
+        m_color_attachment.setStencilStoreOp( vk::AttachmentStoreOp::eDontCare );
+        m_color_attachment.setInitialLayout( vk::ImageLayout::eUndefined );
+        m_color_attachment.setFinalLayout( vk::ImageLayout::ePresentSrcKHR );
+
+        return static_cast<Derived&>( *this );
+    };
+
+    template <typename SubpassDep> void withSubpassDependencies( SubpassDep&& dependecies )
+    {
+        m_subpass_dependecies = ranges::views::all( dependecies ) | ranges::to_vector;
+    }
+
+    Derived& withRenderPass( vk::Device device ) &
+    {
+        vk::AttachmentReference color_attachment_ref{
+            .attachment = 0,
+            .layout = vk::ImageLayout::eColorAttachmentOptimal };
+
+        vk::SubpassDescription subpass{
+            .pipelineBindPoint = vk::PipelineBindPoint::eGraphics,
+            .colorAttachmentCount = 1,
+            .pColorAttachments = &color_attachment_ref };
+
+        vk::RenderPassCreateInfo render_pass_create_info{
+            .attachmentCount = 1,
+            .pAttachments = &m_color_attachment,
+            .subpassCount = 1,
+            .pSubpasses = &subpass,
+            .dependencyCount = static_cast<uint32_t>( m_subpass_dependecies.size() ),
+            .pDependencies = m_subpass_dependecies.data() };
+
+        m_render_pass = device.createRenderPassUnique( render_pass_create_info );
+
+        return static_cast<Derived&>( *this );
+    }
+
+    vk::RenderPass getRenderPass() const { return m_render_pass.get(); }
+
+    void make( vk::GraphicsPipelineCreateInfo& pipeline_create_info )
+    {
+        pipeline_create_info.setRenderPass( getRenderPass() );
+        pipeline_create_info.setSubpass( 0 );
+    }
+
+  protected:
+    RenderPassCfg() = default;
+    RenderPassCfg( RenderPassCfg&& ) = default;
+    RenderPassCfg& operator=( RenderPassCfg&& ) = default;
+    // deleted because of vk::UniqueRenderPass
+    RenderPassCfg( const RenderPassCfg& ) = delete;
+    RenderPassCfg& operator=( const RenderPassCfg& ) = delete;
+    ~RenderPassCfg() = default;
```

Comment:
In this case ```= default``` would mean deleted because of a non-moveable member

Reply:
Well, yes but it will generate warning as I know. Anyway I prefer to be explicit here